Nos/Views/Components/FlagOptionPicker.swift

```diff
@@ -0,0 +1,108 @@
+import SwiftUI
+/// Displays a list of selectable flag options
+struct FlagOptionPicker: View {
+    @Binding private var selectedOption: FlagOption?
+    var options: [FlagOption]
+    var title: String
+    var subTitle: String?
+
+    init(selectedOption: Binding<FlagOption?>, options: [FlagOption], title: String, subTitle: String?) {
+        self._selectedOption = selectedOption
+        self.options = options
+        self.title = title
+        self.subTitle = subTitle
+    }
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            HeaderView(text: title)
+            
+            if let subTitle = subTitle {
+                HeaderView(text: subTitle)
+            }
+            flagOptionsListView
+        }
+        .padding()
+    }
+
+    private var flagOptionsListView: some View {
+        VStack(alignment: .leading) {
+            ForEach(options) { flag in
+                FlagPickerRow(flag: flag, selection: $selectedOption)
+                BeveledSeparator()
+            }
+        }
+        .background(LinearGradient.cardBackground)
+        .clipShape(RoundedRectangle(cornerRadius: 15))
+    }
+}
+
+/// A single row  for a single flag option
+struct FlagPickerRow: View {
+    var flag: FlagOption
+    @Binding var selection: FlagOption?
+
+    var isSelected: Bool {
+        selection?.id == flag.id
+    }
+
+    var body: some View {
+        HStack(alignment: .center) {
+            VStack(alignment: .leading, spacing: 8) {
+                Text(flag.title)
+                    .foregroundColor(.primaryTxt)
+                    .font(.clarity(.regular))
+
+                if let description = flag.description {
+                    Text(description)
+                        .foregroundColor(.secondaryTxt)
+                        .font(.clarity(.regular, textStyle: .footnote))
+                        .lineSpacing(8)
+                }
+            }
+
+            Spacer()
+
+            Button {
+                selection = flag
+            } label: {
+                NosRadioButton(isSelected: isSelected)
+            }
+        }
+        .padding(12)
+    }
+}
+
+private struct HeaderView: View {
+    var text: String
+    var body: some View {
+        Text(text)
+            .lineSpacing(5)
+            .foregroundColor(.primaryTxt)
+            .font(.clarity(.bold))
+            .padding(.bottom, 28)
+    }
+}
+
+struct FlagOptionPicker_Previews: PreviewProvider {
```

Comment:
Would you be able to convert this to the newer `#Preview`? I see you used it in other files, so maybe there was an issue with this one.